tinygrad/codegen/linearizer.py

```diff
@@ -64,6 +64,10 @@ def get_reduce_acc(self, op, dtype:DType):
     AndNode: lambda self,ops,ctx:
       functools.reduce(lambda a,b: ctx.uop_alu_idx(a, b, ops, ctx, BinaryOps.MUL, dtype=dtypes.bool), self.nodes[1:], self.nodes[0].render(ops,ctx)) }
 
+  def get_uop_dtype(self, dtype: DType, amt=1):
```

Comment:
so this function has nothing to do with `uop_dtype`. it's either extracting the base from `ImageDType` or applying `amt` to make a `vec`

something like
```
dtype = buf.dtype.base if isinstance(buf.dtype, ImageDType) else buf.dtype
if amt > 1: dtype = dtype.vec(amt)
```
reads better

Reply:
we should duplicate the same logic in the localtype def and also maybe in the uop function or somewhere else, that's why I wanna isolate it

Reply:
for cases like this i'd make a function when we are really reusing it, otherwise later we might find it's premature optimization. also it has very little to do with unary op check and can be a separate refactor pr